.github/workflows/unit-test.yml

```diff
@@ -0,0 +1,70 @@
+name: "Unit Tests"
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+  push:
+    branches:
+      - master
+      - "v[0-9]+.[0-9]+"
+    tags:
+      - "(dev-)?v[0-9]+.[0-9]+.[0-9]+"
+
+jobs:
+  define-versions:
+    runs-on: ubuntu-latest
+    outputs:
+      fdiVersions: ${{ steps.versions.outputs.fdiVersions }}
+      cdiVersions: ${{ steps.versions.outputs.cdiVersions }}
+      pyVersions: '["3.8", "3.9", "3.10", "3.11", "3.12", "3.13"]'
+    steps:
+      - uses: actions/checkout@v4
+      - uses: supertokens/get-supported-versions-action@main
+        id: versions
+        with:
+          has-fdi: true
+          has-cdi: true
+
+  test:
+    runs-on: ubuntu-latest
+    needs: define-versions
+    strategy:
+      fail-fast: false
+      matrix:
+        py-version: ${{ fromJSON(needs.define-versions.outputs.pyVersions) }}
+        cdi-version: ${{ fromJSON(needs.define-versions.outputs.cdiVersions) }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: supertokens/get-versions-action@main
+        id: versions
+        with:
+          driver-name: python
+          cdi-version: ${{ matrix.cdi-version }}
+        env:
+          SUPERTOKENS_API_KEY: ${{ secrets.SUPERTOKENS_API_KEY }}
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.py-version }}
+      - name: Create virtual environment and install dependencies
```

Comment:
It's not critical at all, but please check if we could cache this based on makefile/requirements hashes.

Reply:
Github has a few actions to do this: `actions/cache/restore@v4` and `actions/cache/save@v4`.

A point of concern in Python is that there is no equivalent of a `package-lock.json` which pins exact versions, so we end up installing the latest available versions. Caching could result in us not using the latest versions of standard libraries like `setuptools`, which we don't explicitly depend on.

There does not seem to be a way to set a timeout on the cache, but it should be possible to clean up manually with a scheduled action (like a cron). The other option is to manually clean up caches.

The current install step takes ~30s, so the cached version should be able to bring it down to atleast ~5s.
How would you like to approach this?

Reply:
No need to implement right now, lets just keep in mind for later.